mars/learn/contrib/pytorch/dataset.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+import numpy as np
+import pandas as pd
+try:
+    import torch
+    from torch.utils.data import Dataset
+except ImportError: # pragma: no cover
+    torch = None
+    Dataset = object
+
+from ....core.context import get_context
+from ....tensor.core import TENSOR_TYPE
+from ....dataframe.core import DATAFRAME_TYPE, SERIES_TYPE
+from ....utils import require_not_none
+
+
+@require_not_none(torch)
+class MarsDataset(Dataset):
+    r"""MarsDataset that inherit from torch.utils.data.Dataset.
+    It converts from Mars basic datatype such as Tensor,
+    DataFrame, Series. Additionally, it's constructor can receive
+    np.ndarray, pd.DataFrame, pd.Series type.
+    """
+    def __init__(self, *tileables, fetch_kwargs=None):
+
+        self._context = get_context()
+        self._tileables = tileables
+        self._fetch_kwargs = fetch_kwargs or dict()
+
+        self._check_and_execute()
+
+    def _check_and_execute(self):
+        for t in self._tileables:
+            if isinstance(t, TENSOR_TYPE):
+                t.execute()
+            elif isinstance(t, DATAFRAME_TYPE):
+                t.execute()
+            elif isinstance(t, SERIES_TYPE):
+                t.execute()
```

Comment:
Add an `else` to raise TypeError if uknown types show up.

Reply:
OK, I have add a type_check. Please check it!